packages/core/src/model.ts

```diff
@@ -139,9 +139,54 @@ export class ChartModel {
 			dataGroups
 		});
 
+		if (this.getZoomBarData()) {
+			// if we have zoom bar data we need to update it as well
+			this.setZoomBarData();
+		}
 		return sanitizedData;
 	}
 
+	/**
+	 * @param zoomBarData any special zoom bar data to use instead of the model data
+	 */
+	setZoomBarData(newZoomBarData?) {
+		const { cartesianScales } = this.services;
+		const domainIdentifier = cartesianScales.getDomainIdentifier();
+		const rangeIdentifier = cartesianScales.getRangeIdentifier();
+		const sanitizedData = newZoomBarData
+			? this.sanitize(Tools.clone(newZoomBarData))
+			: this.getDisplayData(); // if we're not passed explicit zoom data use the model
+
+		// get all dates (Number) in displayData
+		let allDates = sanitizedData.map((datum) =>
+			datum[domainIdentifier].getTime()
+		);
+		allDates = Tools.removeArrayDuplicates(allDates).sort();
+
+		// Go through all date values
+		// And get corresponding data from each dataset
+		const zoomBarNormalizedValues = allDates.map((date) => {
+			let sum = 0;
+			const datum = {};
+
+			sanitizedData.forEach((data) => {
+				if (data[domainIdentifier].getTime() === date) {
+					sum += data[rangeIdentifier];
+				}
+			});
+			datum[domainIdentifier] = new Date(date);
```

Comment:
do we still need to wrap date values into JS-dates? The sanitize PR should've addressed this and now you should be getting JS-dates already

Reply:
I do, because in this case the date is actually the UTC time (see line 162)